### PR TITLE
ColladaLoader: Improve evaluation of transparency

### PR DESCRIPTION
**Pull request by Mugen87**
Solves #13207

### CODE REVIEW

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
I think you should only set `material.transparent = true` if it is requested, or if `material.opacity < 1`.

File: examples/js/loaders/ColladaLoader.js [hunk 1]
```diff
@@ -1471,11 +1476,24 @@ THREE.ColladaLoader.prototype = {
...
+
+								var opaque = parameters[ 'transparent' ].opaque;
+
+								if ( opaque === 'RGB_ZERO' ) material.opacity = 1 - material.opacity;
+
+							}
+
+						}
+
 						material.transparent = true;
```
Comment (Mugen87) [inline examples/js/loaders/ColladaLoader.js]: > I think you should only set material.transparent = true if it is requested, or if material.opacity < 1.

So change this line to:
```javascript
if ( material.opacity < 1 ) material.transparent = true;
```
I guess we can also remove the `material.transparent = true;` statement for the  case `transparent`, see https://github.com/Mugen87/three.js/blob/0e59028607982a876169adfe16fea591faf67ed7/examples/js/loaders/ColladaLoader.js#L1480

Comment (WestLangley) [reply examples/js/loaders/ColladaLoader.js]: I would be guessing, so do your best. :)

Comment (Mugen87) [reply examples/js/loaders/ColladaLoader.js]: see 2a13b83391b14fc08c047815063bf59c776b66af

**Comment by mrdoob** [comment]
I just bumped into a model that has this:

```xml
            <transparent opaque="RGB_ZERO">
              <color>0 0 0 1</color>
            </transparent>
            <transparency>
              <float>1</float>
            </transparency>
```

OSX viewer produces this:

screen shot 2018-02-05 at 4 29 54 pm

But `ColladaLoader` produces `material.opacity = 0` and `material.transparent = true`:

screen shot 2018-02-05 at 4 33 00 pm

https://twitter.com/daitomanabe/status/960500073183379456

**Comment by Mugen87** [comment]
I'll make a PR for this. There is also an other issue with the model:
```
THREE.ColladaLoader: Geometry has faces with more than 4 vertices.
```
But that is actually a known limitation.

**Comment by Mugen87** [comment]
#13260

**Pull request merged**